# module09-penghy27
module09-penghy27 created by GitHub Classroom

### module09-penghy27
### Date: April 2nd, 2023
### Overview:
Run metagenomics analysis with the library metagenomeSeq, follow the tutorial  https://bioconductor.org/packages/release/bioc/vignettes/metagenomeSeq/inst/doc/metagenomeSeq.pdf and run each chunk of R code until 4.1.1.

### References
Paulson, Joseph N., O. Colin Stine, Héctor Corrada Bravo, and Mihai Pop. 2013. “Differential Abundance Analysis for Microbial Marker-Gene Surveys.” *Nature Methods* 10 (12): 1200.

Quince, Christopher, Alan W. Walker, Jared T. Simpson, Nicholas J. Loman, and Nicola Segata. 2017. “Shotgun Metagenomics, from Sampling to Analysis.” *Nature Biotechnology* 35 (9): 833..

Knight, R., Vrbanac, A., Taylor, B.C. et al. Best practices for analysing microbiomes. Nat Rev Microbiol 16, 410–422 (2018). https://doi.org/10.1038/s41579-018-0029-9
